### PR TITLE
Add contract setup and signature verification on the maker side

### PR DESCRIPTION
**Pull request by klochowicz**
Extract the functionality into its own module, as the logic was exactly the same
on both sides.

I've introduced 2 new concepts:
     - "own" - the caller of the function, it's either taker or maker,
        depending on the OwnParams parameter,
     - "other" - the other party, the opposite role.

XXX: Should I rather follow the naming convention in Origin enum (Ours/Theirs?)

### CODE REVIEW

File: daemon/src/contract.rs
Comment (thomaseizinger) [inline daemon/src/contract.rs]: Only half a sentence :)

File: daemon/src/contract.rs
Comment (thomaseizinger) [inline daemon/src/contract.rs]: As a general theme, try to put the public API of a module at the very top and order things from highest to lowest level of importance. That way, as you start reading the code, you know what this module does and by continuing to read, you will learn about the details how it is implemented. Assuming descriptive names, just reading the top-level public function may give you enough information about what this is doing without having to scan down further!

File: daemon/src/contract.rs
Comment (thomaseizinger) [inline daemon/src/contract.rs]: To align this with our naming, we could call this `new` and the file `setup_contract_actor`.

**Review by thomaseizinger — APPROVED** [review]
Nice! I was tempted to do this yesterday but wanted to wait until everything is merged :D

Thanks for unifying this!

A few nits but otherwise LGTM!

File: daemon/src/contract.rs
Comment (klochowicz) [reply daemon/src/contract.rs]: oops! 😳 thanks for catching that!

File: daemon/src/contract.rs
Comment (klochowicz) [reply daemon/src/contract.rs]: that makes a lot of sense! will change things around to reflect this.

Comment (klochowicz) [reply daemon/src/contract.rs]: @thomaseizinger I've read somewhere (in the rustbook?) that it's customary for `const` to  be at the top of the file - shall I keep the `REFUND_TRESHOLD` above the `fn new()` ?

Comment (thomaseizinger) [reply daemon/src/contract.rs]: Yeah consts can stay at the top :)

**Review by thomaseizinger — APPROVED** [review]
Thanks!

File: daemon/src/setup_contract_actor.rs
Comment (bonomat) [inline daemon/src/setup_contract_actor.rs]: Nit: since this function is called by both parties now, it's not `maker: PartyParams` and `taker_cets` but `other: PartyParams` and `own_cets...`
Same goes for the usage of `maker_encsig` in the function body.

**Review by bonomat — APPROVED** [review]
Looks good. Nit on some parameter naming which could be changed, but otherwise, it looks identical to what's been before :)

File: daemon/src/setup_contract_actor.rs
Comment (klochowicz) [reply daemon/src/setup_contract_actor.rs]: indeed, I missed that one - fixed!

**Pull request merged**
